src/components/generics/vacation-test-forms/my-vacation-requests.tsx

```diff
@@ -65,6 +71,70 @@ const RenderVacationRequests = () => {
     initializeRequests();
   }, [person]);
 
+  /**
+   * Initializes all vacation request statuses
+   */
+  const initializeRequestStatuses = async () => {
+    if (!person) return;
+
+    try {
+      const vacationRequestStatuses: VacationRequestStatus[] = [];
+
+      const statusesApi = Api.getVacationRequestStatusApi(accessToken?.access_token);
+
+      await Promise.all(requests.map(async request => {
+        const createdStatuses = await statusesApi.listVacationRequestStatuses({ id: request.id! });
+        createdStatuses.forEach(createdStatus => {
+          vacationRequestStatuses.push(createdStatus);
+        });
+      }));
+
+      setStatuses(vacationRequestStatuses);
+    } catch (error) {
+      context.setError(strings.errorHandling.fetchVacationDataFailed, error);
+    }
+  };
+
+  useEffect(() => {
+    if (requests.length <= 0) {
+      return;
+    }
+    initializeRequestStatuses();
+  }, [requests]);
+
+  /**
+   * Initializes all the latest vacation request statuses, so there would be only one status for each request showed on the UI
```

Comment:
Again i'm seeing duplication here, can this be avoided- don't worry if not, there is quite a bit of duplication in the project already and this is part of the reason why we will be rebuilding the project in the future.

Reply:
decided on leaving the duplicates in, won't leave duplicates in the new project